sponsorship_compassion/model/contracts.py

```diff
@@ -132,125 +151,93 @@ def _get_type(self, cr, uid, context=None):
         return res
 
     ##########################################################################
-    #                                 FIELDS                                 #
-    ##########################################################################
-    _columns = {
-        'correspondant_id': fields.many2one(
-            'res.partner', _('Correspondant'), required=True, readonly=True,
-            states={'draft': [('readonly', False)],
-                    'waiting': [('readonly', False)],
-                    'mandate': [('readonly', False)]},
-            track_visibility='onchange'),
-        'partner_codega': fields.related(
-            'correspondant_id', 'ref', string=_('Partner ref'), readonly=True,
-            type='char'),
-        'fully_managed': fields.function(
-            _is_fully_managed, type="boolean", store={
-                'recurring.contract': (
-                    lambda self, cr, uid, ids, c=None: ids,
-                    ['partner_id', 'correspondant_id'], 10)
-            }),
-        'birthday_invoice': fields.float(_("Annual birthday gift"), help=_(
-            "Set the amount to enable automatic invoice creation each year "
-            "for a birthday gift. The invoice is set two months before "
-            "child's birthday."), track_visibility='onchange'),
-    }
-
-    _defaults = {
-        'contract_line_ids': _get_standard_lines
-    }
-
-    ##########################################################################
     #                              ORM METHODS                               #
     ##########################################################################
-    def create(self, cr, uid, vals, context):
+
+    @api.model
+    def create(self, vals):
         """ Perform various checks on contract creations
         """
-        child_id = vals.get('child_id')
-        if 'S' in vals.get('type', '') and child_id:
-            self.pool.get('compassion.child').write(
-                cr, uid, child_id, {
-                    'sponsor_id': vals['partner_id']}, context)
+        child = self.env['compassion.child'].browse(vals.get('child_id'))
+        if 'S' in vals.get('type', '') and child:
+            child.write({'sponsor_id': vals['partner_id']})
 
         if 'group_id' in vals:
-            if 'S' in context.get('default_type', 'O'):
+            if 'S' in self.env.context.get('default_type', 'O'):
                 group_id = vals['group_id']
-                if not self._is_a_valid_group(cr, uid, group_id, context):
-                    raise orm.except_orm(
+                if not self._is_a_valid_group(group_id):

```

Comment:
I think for these kinds of checks in create or write methods, the new way would be to do it in a @api.constrains decorated method.